internal/processmon/processmon.go

```diff
@@ -38,107 +33,117 @@ var (
 	ErrFailedtoLaunch = errors.New("Failed to launch enforcer")
 	// ErrProcessDoesNotExists Exported
 	ErrProcessDoesNotExists = errors.New("Process in that context does not exist")
-	//ErrBinaryNotFound Exported
+	// ErrBinaryNotFound Exported
 	ErrBinaryNotFound = errors.New("Enforcer Binary not found")
 )
 
 const (
+	// netnspath holds the directory to ensure ip netns command works
+	netnspath               = "/var/run/netns/"
 	processMonitorCacheName = "ProcessMonitorCache"
 	secretLength            = 32
 )
 
-//ProcessMon exported
-type ProcessMon struct {
+// processMon is an instance of processMonitor
+type processMon struct {
+	// netnspath made configurable to enable running tests
+	netnspath       string
 	activeProcesses *cache.Cache
+	childExitStatus chan exitStatus
+	// logToConsole stores if we should log to console
+	logToConsole bool
+	// launcProcessArgs are arguments that are provided to all processes launched by processmon
+	launcProcessArgs []string
 }
 
-//ProcessInfo exported
+// processInfo stores per process information
 type processInfo struct {
 	contextID string
 	RPCHdl    rpcwrapper.RPCClient
 	process   *os.Process
-	deleted   bool
 }
 
-//ExitStatus captures the exit status of a process
-//The contextID is optional and is primarily used by remote enforcer processes
-//and represents the namespace in which the process was running
-type ExitStatus struct {
-	process    int
+// exitStatus captures the exit status of a process
+type exitStatus struct {
+	process int
+	// The contextID is optional and is primarily used by remote enforcer
+	// processes to represent the namespace in which the process was running
 	contextID  string
 	exitStatus error
 }
 
 func init() {
-
-	netnspath = "/var/run/netns/"
-
-	go collectChildExitStatus()
-
 	// Setup new launcher
-	newProcessMon()
+	newProcessMon(netnspath)
 }
 
-//collectChildExitStatus is an async function which collects status for all launched child processes
-func collectChildExitStatus() {
+// contextID2SocketPath returns the socket path to use for a givent context
+func contextID2SocketPath(contextID string) string {
 
-	for {
-		exitStatus := <-childExitStatus
-		zap.L().Debug("Remote enforcer exited",
-			zap.String("nativeContextID", exitStatus.contextID),
-			zap.Int("pid", exitStatus.process),
-			zap.Error(exitStatus.exitStatus),
-		)
-	}
+	return filepath.Join("/var/run/" + contextID + ".sock")
```

Comment:
maybe this function should check for the validity of the given contextID (if it's empty for instance), also that there are no `..` etc.. and return either an error to tell people, or it that should never ever happen, then panics